spring-app/src/main/java/pl/edu/agh/cqm/controller/SingleCdnMonitoringController.java

```diff
@@ -0,0 +1,115 @@
+    ) {
+        try {
+            Map<String, List<RTTSampleDTO>> rttSamples = monitoringService.getRTTSamples(searchDTO.getCdn(),
+                    searchDTO.getStartDate(), searchDTO.getEndDate(), searchDTO.getGranularity());
+            if (rttSamples.values().stream().allMatch(List::isEmpty)) {
+                throw new IllegalArgumentException();
+            }
+
+            return new SingleCdnSingleParameterResponseDTO<>(
+                    searchDTO.getCdn(),
+                    searchDTO.getStartDate(),
+                    searchDTO.getEndDate(),
+                    rttSamples,
+                    deviationsService.getRTTDeviations(rttSamples),
+                    parameterService.getParameterHistory(searchDTO.getStartDate(), searchDTO.getEndDate())
+            );
+        } catch (IllegalArgumentException e) {
+            throw new BadRequestException();
+        }
+    }
+
+    @GetMapping("/throughput")
+    public SingleCdnSingleParameterResponseDTO<ThroughputSampleDTO> getThroughput(
+            @Valid SingleCdnSampleSearchDTO searchDTO
+    ) {
+        try {
+            Map<String, List<ThroughputSampleDTO>> throughputSamples = monitoringService.getThroughputSamples(
+                    searchDTO.getCdn(), searchDTO.getStartDate(), searchDTO.getEndDate(), searchDTO.getGranularity());
+            if (throughputSamples.values().stream().allMatch(List::isEmpty)) {
+                throw new IllegalArgumentException();
+            }
+
+            return new SingleCdnSingleParameterResponseDTO<>(
+                    searchDTO.getCdn(),
+                    searchDTO.getStartDate(),
+                    searchDTO.getEndDate(),
+                    throughputSamples,
+                    deviationsService.getThroughputDeviations(throughputSamples),
+                    parameterService.getParameterHistory(searchDTO.getStartDate(), searchDTO.getEndDate())
+            );
+        } catch (IllegalArgumentException e) {
+            throw new BadRequestException();
+        }
+    }
+
+    @GetMapping("/all")
+    public SingleCdnAllParametersResponseDTO getAll(
+            @Valid SingleCdnSampleSearchDTO searchDTO
+    ) {
+        try {
+            String cdn = searchDTO.getCdn();
+            Instant startDate = searchDTO.getStartDate();
+            Instant endDate = searchDTO.getEndDate();
+            Long granularity = searchDTO.getGranularity();
+
+            Map<String, List<RTTSampleDTO>> rttSamples =
+                    monitoringService.getRTTSamples(cdn, startDate, endDate, granularity);
+            Map<String, List<ThroughputSampleDTO>> throughputSamples =
+                    monitoringService.getThroughputSamples(cdn, startDate, endDate, granularity);
+
+            if (rttSamples.values().stream().allMatch(List::isEmpty)
+                    && throughputSamples.values().stream().allMatch(List::isEmpty)) {
+                throw new IllegalArgumentException();
```

Comment:
we could just throw BadRequestException here and remove try catch

Reply:
applies to all 3 endpoints

Reply:
`IllegalArgumentException` can be also thrown when there is no cdn in the database with name `cdn` from the query. (https://github.com/ioagh2020d/CDN-Quality-Monitor/blob/6000c0af286c8448a361415a1cd34cade74d00ca/spring-app/src/main/java/pl/edu/agh/cqm/service/impl/ParameterServiceImpl.java#L148)